src/components/TodoList/TodoList.tsx

```diff
@@ -1,4 +1,24 @@
-// Add the required types and props
-export const TodoList = () => (
-  <>TodoList markup</>
-);
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoInfo } from '../TodoInfo';
+
+type Props = {
+  todos: Todo[];
+};
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
+  const todoList
+  = todos.length > 0 ? (
+    <ul>
+      {todos.map((todo) => (
+        <li key={todo.id}>
+          <TodoInfo todo={todo} />
+        </li>
+      ))}
+    </ul>
+  ) : (
+    ''
+  );
```

Comment:
You don't need to render this component if you don't have todos at all
```suggestion
    <ul>
      {todos.map((todo) => (
        <li key={todo.id}>
          <TodoInfo todo={todo} />
        </li>
      ))}
    </ul>
```

Just add a conditional rendering 
```
{todos.length && <TodoList todos={todos} />}

Reply:
Done